src/main/java/com/github/slavaz/maven/plugin/postgresql/embedded/goals/AbstractGoalMojo.java

```diff
@@ -1,25 +1,91 @@
 package com.github.slavaz.maven.plugin.postgresql.embedded.goals;
 
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.IterableUtils;
+import org.apache.commons.collections4.Predicate;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
 
 /**
  * Created by slavaz on 13/02/17.
  */
 public abstract class AbstractGoalMojo extends AbstractMojo {
 
-    @Parameter(defaultValue = "false")
-    private boolean skipGoal;
+	protected static final String SP_HTTP_PROXY_HOST = "http.proxyHost";
+
+	protected static final String SP_HTTP_PROXY_PORT = "http.proxyPort";
+
+	protected static final String SP_HTTP_NO_PROXY = "http.nonProxyHosts";
+
+	protected static final String PROTOCOL_HTTP = "http";
+
+	@Parameter(defaultValue = "false")
+	private boolean skipGoal;
+
+	@Parameter(defaultValue = "${settings}", readonly = true)
+	protected Settings settings;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skipGoal) {
+			getLog().debug("Skipped.");
+		} else {
+			doExecute();
+		}
+	}
+
+	protected void doExecute() throws MojoExecutionException, MojoFailureException {
+		handleProxyConfigurjation();
+	}
+
+	/**
+	 * Thie method sets the proper system properties to force http connections to use proxy when downloading Postgres binaries
+	 * If system property http.proxyHost is already set, the proxies defined on settings are not taken in consideration
+	 * Authentication is taken in consideration using {@link Authenticator#setDefault(Authenticator)}
+	 */
+	protected void handleProxyConfigurjation() {
```

Comment:
I've refactored all proxy configuration outside from Abstract Mojo to a new helper class ProxyUtils